tests/ResourceSyncingTest.php

```diff
@@ -1173,6 +1198,53 @@
     expect($centralUser->foo)->toBe('bar');
 });
 
+test('resource syncing works correctly when using a global scope on a tenant model', function(bool $scopeGetModelQuery) {
+    if ($scopeGetModelQuery) {
+        UpdateOrCreateSyncedResource::$scopeGetModelQuery = function (Builder $query) {
+            if ($query->getModel()->hasGlobalScope(TestingScope::class)) {
+                $query->withoutGlobalScope(TestingScope::class);
+            }
+        };
+    }
+
+    $centralUser = CentralUser::firstOrCreate(
```

Comment:
No idea why we're using ambiguous code like `firstOrCreate` when we have full control over the context in this test. If the user already exists, use it. If it doesn't, create it. **We know which one it is** in tests. In this case the latter, there's no user created in `beforeEach()`.